examples/migration/migrate.py

```diff
@@ -0,0 +1,300 @@
+
+    def get_migratable_device(self):
+        '''Method to get list of migratable device .
+
+        Return
+        ------
+        list containing server hardware that are eligible for migration:
+
+        '''
+
+        url = '/rest/migrate/migratable-server-hardware/'
+        payload = {"ipAddress": self.migrate_source_ip,
+                   "username": self.migrate_source_username, "password": self.migrate_source_password}
+
+        (migrateable_success, migrateable_resp, migrateable_json_response, _) = self.appliance_request(
+            request_type='POST', url=url, secure=True, payload=payload)
+        if migrateable_success:
+            logging.info('migrateable accepted.')
+        else:
+            raise Exception('migratable device listing failed. Status: {0}. JSON Response: {1}'
+                            .format(migrateable_resp.status_code, json.dumps(migrateable_json_response, sort_keys=True,
+                                                                             indent=4, separators=(',', ': '))))
+
+        sh_list = migrateable_json_response['serverHardware']
+
+        return sh_list
+
+    def do_migrate(self):
+        '''Method to perform the migration of selected harware .
+
+        '''
+        url = '/rest/migrate'
+        payload = {"ipAddress": self.migrate_source_ip, "username": self.migrate_source_username,
+                   "password": self.migrate_source_password, "uri": self.migrate_source_hardware_uri}
+
+        migrate_json_response = {}
+        (migrate_success, migrate_resp, migrate_json_response, _) = self.appliance_request(
+            request_type='POST', url=url, secure=True, payload=payload)
+        if migrate_success:
+            logging.info('migration successful.')
+        else:
+            raise Exception('migration failed. Status: {0}. JSON Response: {1}'.
+                            format(migrate_resp.status_code, json.dumps(migrate_json_response, sort_keys=True, indent=4, separators=(',', ': '))))
+
+
+config = {
+    "appliance_ip": "<appliance_ip>",
+    "source_appliance_ip": "<source_appliance_ip>",
+    "api_version": 4000,
+    "credentials": {
+        "userName": "<appliance_username>",
+        "authLoginDomain": "",
+        "password": "<appliance_password>",
+        "sessionID": "",
+        "source_username": "<source_appliance_username>",
+        "source_password": "<source_appliance_password>"
+    }
+}
+
+if __name__ == '__main__':
+
+    if len(sys.argv) >= 2:
+        sh_user = str(sys.argv[1])
+    else:
+        print("input list of server hardware to be migrated")
+        sys.exit(1)
+
+    sh_user_list = sh_user.strip('[]').split(',')
+
+    config = try_load_from_file(config)
+    max_retries_in_session = 10
+
+    serverH = ServerHardware(config)
+
+    sh_list = serverH.get_migratable_device()
+
+    if len(sh_list) <= 0:
+        sys.exit()
```

Comment:
Can you put proper error message why it is exiting..also i think instead of sys.exit() it should be sys.exit(1)